cmd/render/render.go

```diff
@@ -0,0 +1,128 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package render
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/ghodss/yaml"
+	performancev2 "github.com/openshift-kni/performance-addon-operators/api/v2"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/kubeletconfig"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/machineconfig"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/runtimeclass"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/tuned"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/klog"
+)
+
+type renderOpts struct {
+	ppInputFilePath string
+	assetsDir       string
+	outputDir       string
+}
+
+type manifestable map[string]interface{}
+
+//NewRenderCommand creates a render command.
+func NewRenderCommand() *cobra.Command {
+	renderOpts := renderOpts{}
+
+	cmd := &cobra.Command{
+		Use:   "render",
+		Short: "Render performance-addon-operator CR manifests",
+		Run: func(cmd *cobra.Command, args []string) {
+
+			if err := renderOpts.Validate(); err != nil {
+				klog.Error(err)
+			}
+
+			if err := renderOpts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	renderOpts.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (r *renderOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&r.ppInputFilePath, "performance-profile-file-path", r.ppInputFilePath, "Input path for the Performance Profile manifest.")
+	fs.StringVar(&r.assetsDir, "assets-dir-path", components.AssetsDir, "Input path for the assets directory.")
+	fs.StringVar(&r.outputDir, "output-dir", r.outputDir, "Output path for the rendered manifests.")
+}
+
+func (r *renderOpts) Validate() error {
+	if len(r.ppInputFilePath) == 0 || len(r.outputDir) == 0 {
+		return fmt.Errorf("performance-profile-file-path and output-dir must be specified")
+	}
+	return nil
+}
+
+func (r *renderOpts) Run() error {
+	b, err := ioutil.ReadFile(r.ppInputFilePath)
+	if err != nil {
+		return err
+	}
+
+	profile := &performancev2.PerformanceProfile{}
+	err = yaml.Unmarshal(b, profile)
+	if err != nil {
+		return err
+	}
+
+	components := make(manifestable)
+
+	mc, err := machineconfig.New(r.assetsDir, profile)
+	if err != nil {
+		return err
+	}
+	components[mc.Kind] = mc
+
+	kc, err := kubeletconfig.New(profile)
+	if err != nil {
+		return err
+	}
+	components[kc.Kind] = kc
+
+	performanceTuned, err := tuned.NewNodePerformance(r.assetsDir, profile)
+	if err != nil {
+		return err
+	}
+	components[performanceTuned.Kind] = performanceTuned
+
+	runtimeClass := runtimeclass.New(profile, machineconfig.HighPerformanceRuntime)
+	components[runtimeClass.Kind] = runtimeClass
+
+	for name, component := range components {
```

Comment:
not sure the kind is a great fit for the name

Reply:
The names of the manifests are ```Kind.yaml``` so maybe instead of name call it ```fileName``` ?

Reply:
let's keep like this for the time being and re-evaluate later.